ios/brave-ios/Sources/DataImporter/UI/DataImporterStateView.swift

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import BraveStrings
+import BraveUI
+import DesignSystem
+import SwiftUI
+
+struct DataImporterStateView: View {
+  typealias Action = () -> Void
+
+  @Environment(\.dismiss)
+  private var dismiss
+
+  var kind: Kind
+  var model: DataImportModel
+
+  var primaryAction: Action
+  var secondaryAction: Action?
+
+  var body: some View {
+    VStack {
+      if kind == .passwordConflict {
+        Button {
+          dismiss()
+        } label: {
+          Label {
+            Text(Strings.close)
+          } icon: {
+            Image(braveSystemName: "leo.close")
+              .foregroundColor(Color(braveSystemName: .iconDefault))
+              .padding(8)
+          }
+          .labelStyle(.iconOnly)
+        }
+        .background(Color(braveSystemName: .materialSeparator), in: Circle())
+        .frame(maxWidth: .infinity, alignment: .trailing)
+      }
```

Comment:
This shouldn't be part of the VStack, it should be in a `overlay(alignment: .topTrailing)`

Reply:
It also probably shouldn't be part of the `ScrollView`

Reply:
This view doesn't have a scroll-view.